### PR TITLE
Support for MaxBytes for pull requests.

### PR DESCRIPTION
**Pull request by derekcollison**
Getting us prepped for move to all pull consumers for clients and for the internal consumers for mirrors/sources.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

**Review by matthiashanel — APPROVED** [review]
LGTM

File: server/jetstream_test.go [hunk 1]
```diff
@@ -17369,6 +17377,114 @@ func TestJetStreamWorkQueueSourceNamingRestart(t *testing.T) {
 	}
 }
 
+func TestJetStreamPullMaxBytes(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer s.Shutdown()
```
Comment (kozlovic) [inline server/jetstream_test.go:17382]: Move the defer shutdown past the defer removeDir please.

**Review by kozlovic — CHANGES_REQUESTED** [review]

File: server/jetstream_test.go (hunk 1 above)
Comment (derekcollison) [reply server/jetstream_test.go]: ok done. I keep copying it from somewhere that is like that IIRC.

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**